### ISSUE TITLE
Use fewer database connections in eWallet connection pool

### ISSUE THREAD

**Issue by sirn** [omisego/ewallet#327]
Currently 3 instances of `ewallet` is using approximately 60 connections (20 each for each instance, 10 for ewallet, 10 for ledger):

```
postgres=> select datname, COUNT(datname) from pg_stat_activity GROUP BY datname;
     datname     | count
-----------------+-------
 staging_ewallet |    30
 staging_ledger  |    30
(4 rows)
```

This might cause problems with some setups, such as Google Cloud SQL which has [very low limit](https://issuetracker.google.com/issues/37271935) by default to prevent OOM.

**Comment by sirn** [comment]
I think we can disable connection pooling for some components e.g. eWallet Config which AFAIK put its value into corresponding application env on start and stays that way until it recieve a message to reload. 

Or maybe using [shared connection pool](
https://elixirforum.com/t/sharing-a-dbconnection-pool-with-multiple-ecto-repos/13832/)3?

**Comment by unnawut** [comment]
I've researched a bit more based on comments above:

- **Approach 1: One repo for the entire umbrella ([ref1](https://elixirforum.com/t/sharing-a-dbconnection-pool-with-multiple-ecto-repos/13832/3), [ref2](https://elixirforum.com/t/sharing-a-dbconnection-pool-with-multiple-ecto-repos/13832/4)):**
  - Pro: Relatively easy to merge at repo level, no low level hacking
  - Con: Less separation of concerns
  - Con: All migrations need to be moved to that single shared repo (unless we hack Ecto's migrator).

- **Approach 2: Implement a custom, connection-sharing `DBConnection.Pool` ([ref3](https://elixirforum.com/t/sharing-a-dbconnection-pool-with-multiple-ecto-repos/13832/2)):**
  - Pro: Can keep the repos and migrations separate (as they are now)
  - Con: Needs a few low-level hacking with Ecto. E.g.
    - Override how Ecto creates connections, which is not public API
    - Database credentials need to be untied from repo level to connection level

Basically I think Ecto is not designed to share connections across repos at all, and rightly so.

---

So I'll be taking the easy way out for now. Picking these suggestions:

1. @mederic-p's suggestion to make the pool size ENV-configurable for each repo.
2. @sirn's suggestion to disable connection pooling on unnecessary modules i.e. `EWalletConfig.Repo`
